specs/interop/predeploys.md

```diff
@@ -416,6 +426,30 @@ dependency set called `dependencySet()`. This function MUST return the array of
 `L1Block` MUST also provide a public getter to get the dependency set size called `dependencySetSize()`. This function
 MUST return the length of the dependency set array.
 
+### Deposit Context
+
+The `L1Block` contract is updated with the following methods to support [deposit contexts](./derivation.md#deposit-context).
+
+#### `isDeposit`
+
+```solidity
+function isDeposit() (return bool)
+```
+
+A new method added to `L1Block` indicating whether the current execution is in a [deposit context](./derivation.md#deposit-context). The method has the following signature:
+
+
+Only the `CrossL2Inbox` is authorized to call `isDeposit`.
+
+#### `setDepositing`
+
+```solidity
+function setDepositing(bool _isDepositing)
+```
+A new method added to indicate whether the current execution is in a deposit context.
+
+Only the `DEPOSITOR_ACCOUNT` is authorized to call `setDepositing`.
```

Comment:
When is this called? For each deposit (assuming `DEPOSITOR_ACCOUNT` is the only party allowed to call it)?

Reply:
It's called twice, at the start of deposits processing and at the end. Rather than for every deposit.

Reply:
What enforces that the calls are made?

Reply:
Derivation is altered to inserts sandwich depositing transactions with `setDepositing` calls. That is, when signaling payload-attribute transactions to execution.
Verifiers would then check that the calls were made, in the correct order.

Reply:
The way that I thought this would work is the `setXXX` would `sstore(isDeposit, true)` and then the deposits would be sandwiched with another system deposit that calls a public `onlyDepositor` function that `sstore(isDeposit, false)`